File: Python/Benchmark.py
import time
from typing import Callable, Any, Dict
from dataclasses import dataclass

class Benchmark:
    def __init__(self, iterations: int = 10000):
        self.iterations = iterations
        self.results = {}

    def measure(self, fn: Callable, params: tuple = ()) -> Dict[str, float]:
        start = time.time()

        for _ in range(self.iterations):
            fn(*params)

        end = time.time()

        elapsed = end - start

        self.results = {
            'time': format(elapsed, '.10f'),
            'iterations': self.iterations,
            'time_per_iteration': format(elapsed / self.iterations, '.10f')
        }
    def print(self):
        print(f"{self.results['time']} s")

File: PHP/Benchmark.php
<?php

class Benchmark
{
    private int $iterations;
    private array $results = [];


    public function __construct (int $iterations = 10000)
    {
        $this->iterations = $iterations;
    }

    public function measure (callable $fn, array $params = []): void
    {
        $start = microtime(true);

        for ($i = 0; $i < $this->iterations; $i++) {
            $fn(...$params);
        }

        $end = microtime(true);

        $elapsed = $end - $start;

        $this->results = [
            'time' => number_format(
                $elapsed,
                10,
                '.',
                ''
            ),
            'iterations' => $this->iterations,
            'time_per_iteration' => number_format(
                $elapsed / $this->iterations,
                10,
                '.',
                ''
            )
        ];
    }
    public function print (): void
    {
        echo "{$this->results['time']} s\n";
    }
}
